ophyd/busy.py

```diff
@@ -0,0 +1,72 @@
+import time
+import threading
+
+from ophyd.device import Device
+from ophyd.status import DeviceStatus
+
+
+class BusyStatus(DeviceStatus):
+    """
+    A "busy" device that takes a fixed amount of time in seconds to report complete.
+
+    The clock starts as soon as the device is created (so do not hold onto these things!)
+
+    Parameters
+    ----------
+    device : Device
+        This object this status object belongs to
+
+    delay : float
+        Total delay in seconds
+
+    tick : float, default=0.1
+        Time between updating the
+
+    """
+
+    def __init__(self, device, delay, *, tick=0.1, **kwargs):
+        super().__init__(device, **kwargs)
+        start = time.monotonic()
+        deadline = start + delay
+
+        def busy_status_loop():
+            current = time.monotonic()
+            while current < deadline:
+                elapsed = current - start
+                for w in self._watchers:
+                    w(
+                        name=self.device.name,
+                        curent=elapsed,
+                        initial=0,
+                        target=delay,
+                        units="s",
```

Comment:
That's what I found as a working example with the correct kwargs: https://github.com/bluesky/ophyd/blob/7937cd46e92e9ec2f7eec321dd5b1b1b71115e20/ophyd/areadetector/trigger_mixins.py#L67-L76
```suggestion
                        unit="s",
```